src/devices/steelmate.c

```diff
@@ -52,46 +55,47 @@ Bytes 2 to 9 are inverted Manchester with swapped MSB/LSB:
 
 static int steelmate_callback(r_device *decoder, bitbuffer_t *bitbuffer)
 {
+    uint8_t const preamble_pattern[] = {0x00, 0x00, 0x7f}; // inverted, raw value is 0x5a
     //Loop through each row of data
-    for (int row = 0; row < bitbuffer->num_rows; row++)
-    {
+    for (int row = 0; row < bitbuffer->num_rows; row++) {
         //Payload is inverted Manchester encoded, and reversed MSB/LSB order
-        uint8_t *b = bitbuffer->bb[row];
+        unsigned row_len = bitbuffer->bits_per_row[row];
 
-        //Length must be 72 bits to be considered a valid packet
-        if (bitbuffer->bits_per_row[row] != 72)
+        //Length must be 72, 73, 208 or 209 bits to be considered a valid packet
+        if (row_len != 72 && row_len != 73 && row_len != 209 && row_len != 208)
             continue; // DECODE_ABORT_LENGTH
 
         //Valid preamble? (Note, the data is still wrong order at this point. Correct pre-amble: 0x00 0x00 0x01)
-        if (b[0] != 0x00 || b[1] != 0x00 || b[2] != 0x7f)
+        unsigned bitpos = bitbuffer_search(bitbuffer, row, 0, preamble_pattern, 24);
+        if (bitpos > row_len - 72)
             continue; // DECODE_ABORT_EARLY
+        bitbuffer_invert(bitbuffer);
+        uint8_t b[9];
+        bitbuffer_extract_bytes(bitbuffer, row, bitpos, b, 72);
 
-        //Preamble
-        uint8_t preamble = ~reverse8(b[2]);
+        reflect_bytes(b, 9);
+
+        //Checksum is a sum of all the other values
+        if ((add_bytes(&b[2], 6) & 0xFF) != b[8])
+            continue; // DECODE_FAIL_MIC
 
         //Sensor ID
-        uint8_t id1 = ~reverse8(b[3]);
-        uint8_t id2 = ~reverse8(b[4]);
+        uint8_t id1 = b[3];
+        uint8_t id2 = b[4];
 
         //Pressure is stored as 32 * the Bar
-        uint8_t p1 = ~reverse8(b[5]);
+        uint8_t p1 = b[5];
 
-        //Temperature is stored in Fahrenheit. Note that the datasheet claims operational to -40'C, but can only express values from -17.8'C
-        uint8_t tempFahrenheit = ~reverse8(b[6]);
+        //Temperature is sent as degrees Celcius + 50.
+        int temp_c = b[6] - 50;
 
-        //Battery voltage is stored as half the mV
-        uint8_t tmpbattery_mV = ~reverse8(b[7]);
-
-        //Checksum is a sum of all the other values
-        uint8_t payload_checksum = ~reverse8(b[8]);
-        uint8_t calculated_checksum = preamble + id1 + id2 + p1 + tempFahrenheit + tmpbattery_mV;
-        if (payload_checksum != calculated_checksum)
-            continue; // DECODE_FAIL_MIC
+        //Battery voltage is stored as 100*(3.9v-<volt>).
+        uint8_t b1     = b[7];
+        int battery_mv = (int)3900 - ((double)b1 * 10.0f);
```

Comment:
All good for merge, though I'd like to mention that this is perhaps not doing what is intended:
`(int)3900` is already int but will be promoted to double because of the right side.
`10.0f` will be promted to double because of the casted b1. 
I guess it would suffice to calc: `int battery_mv = 3900 - b1 * 10;`, right?